crates/mem-pool/src/pool.rs

```diff
@@ -106,8 +112,9 @@ impl MemPool {
         generator: Arc<Generator>,
         provider: Box<dyn MemPoolProvider + Send>,
         error_tx_handler: Option<Box<dyn MemPoolErrorTxHandler + Send>>,
-        config: MemPoolConfig,
+        config: (MemPoolConfig, NodeMode),
```

Comment:
Why use tuple?, we can just add another parameter node_mode: NodeMode.

Reply:
Both are from the config. Plus if we add one more parameter, it would be **7** parameters in total, which triggers the "too many parameters warning". Maybe I can refactor the params by wrapping some of them into a new struct in another pr?

Reply:
IMO, we dont need NodeMode here.
```NodeMode``` is only used in two places: ```push_transaction_with_db``` to skip ```verify_tx()``` and ```notify_new_tip```.

Why not directly call ```finalize_tx()``` instead? and for ```notify_new_tip```, just pass ```None``` to chain in runner.rs.